SPECS/maven/maven.spec

```diff
@@ -0,0 +1,301 @@
+
+%if 0%{?sources_generation}
+%package cached-sources
+Summary:        NOT TO BE USED AS REGULAR PACKAGE! REQUIRES NETWORK ACCESS!
+
+%description cached-sources
+%{summary}
+This is an artificial package to ease generation of cached sources for
+the regular builds of "maven" when the "sources_generation" macro is set to 0.
+
+%endif
+
+%prep
+# Installing 1.0 maven rpm from PMC to provide prebuilt mvn binary.
+rpm -i --nodeps %{SOURCE1}
+which mvn
+
+%if ! 0%{?sources_generation}
+# Setup maven .m2 cache directory
+mkdir /root/.m2
+pushd /root/.m2
+tar xf %{SOURCE2} --no-same-owner
+popd
+%endif
+
+%setup -q -n apache-%{name}-%{version}
+%if ! 0%{?sources_generation}
+# Setup licenses. Remove LICENSE.vm script, which downloads all subproject license files, and replace with prepopulated license tarball.
+rm -v apache-maven/src/main/appended-resources/META-INF/LICENSE.vm
+pushd apache-maven
+tar xf %{SOURCE3} --no-same-owner
+cp -v ./target/licenses/lib/* %{_var}/opt/apache-maven/lib
+popd
+%endif
+
+%build
+# Changing distribution dir to BUILD directory as install macro clears buildroot prior to creating a fresh directory, thus clearing artifacts copied by maven. We copy them later.
+MAVEN_DIST_DIR=%{_builddir}%{_prefixmvn}
+
+export JAVA_HOME="%{_libdir}/jvm/msopenjdk-11"
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$(find $JAVA_HOME/lib -name "jli")
+
+sed -i 's/www.opensource/opensource/g' DEPENDENCIES
+pwd
+echo $LD_LIBRARY_PATH
+echo $MAVEN_DIST_DIR
+mvn -DdistributionTargetDir=$MAVEN_DIST_DIR -DskipTests clean package %{?offline_build}
+
+%check
+echo "1.0 based mvn rpm is uploaded to mariner blob store. Compare checksums of blob rpm and latest pmc rpm package."
+pushd %{_tmppath}
+mvnpmcmd5=
+%ifarch x86_64
+wget https://packages.microsoft.com/cbl-mariner/1.0/prod/base/x86_64/rpms/maven-%{mvn_1_0_pmc_ver}.cm1.x86_64.rpm
+mvnpmcmd5=$(md5sum maven-%{mvn_1_0_pmc_ver}.cm1.x86_64.rpm)
+mvnpmcmd5=$(echo $mvnpmcmd5 | cut -d' ' -f 1)
+%endif
+%ifarch aarch64
+wget https://packages.microsoft.com/cbl-mariner/1.0/prod/base/aarch64/rpms/maven-%{mvn_1_0_pmc_ver}.cm1.aarch64.rpm
+mvnpmcmd5=$(md5sum maven-%{mvn_1_0_pmc_ver}.cm1.aarch64.rpm)
+mvnpmcmd5=$(echo $mvnpmcmd5 | cut -d' ' -f 1)
+%endif
+
+mvnmd5=$(md5sum %{SOURCE1})
+mvnmd5=$(echo $mvnmd5 | cut -d' ' -f 1)
+
+if [[ $mvnpmcmd5 =~ $mvnmd5 ]];then
+echo "mvn m5sum matches."
+else
+echo "Mismatch in mvn md5sum, exiting build."
+popd
+exit 1
+fi
+popd
+
+%install
+mkdir -p %{buildroot}%{_prefixmvn}
+mkdir -p %{buildroot}%{_libdirmvn}
+mkdir -p %{buildroot}%{_bindirmvn}
+
+%if 0%{?sources_generation}
+
+tar -C /root/.m2 -cpvz -f %{m2_cache_tarball_name} repository
+mv %{m2_cache_tarball_name} %{buildroot}%{_prefixmvn}
+
+tar -C apache-maven -cpvz -f %{licenses_tarball_name} target/licenses/lib
```

Comment:
So…if I understand this correctly.. the licenses are downloaded during the build process?  So you have to capture them here?

Reply:
Yes, as per the comments, the License.vm downloads sub-project license files. We need to cache this again due to no download capability.